File: adapter/adapter.py
import csv, json


class CSVAdapter:
    def __init__(self, csv_file):
        self.csv_file = csv_file

    def read_json(self):
        data = []

        with open(self.csv_file) as file:
            csv_reader = csv.DictReader(file)

            for row in csv_reader:
                data.append(row)

        return json.dumps(data)


if __name__ == '__main__':
    csv_adapter = CSVAdapter('data.csv')
    json_data = csv_adapter.read_json()

    print(json_data)

File: adapter/adapter.js
const csv = require('csvtojson');

class CSVAdapter {
    constructor(csvFile) {
        this.csvFile = csvFile;
    }

    async readJson() {
        const data = await csv().fromFile(this.csvFile);

        return JSON.stringify(data);
    }
}

const csvAdapter = new CSVAdapter('data.csv');
csvAdapter.readJson().then(data => {
    console.log(data);
});
